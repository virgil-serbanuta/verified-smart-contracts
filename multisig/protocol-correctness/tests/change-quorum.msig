from address(7) run init(1, [address(7), .]);
from address(7) run proposeAddBoardMember(address(8));
from address(7) run proposeChangeQuorum(2);
from address(7) run sign(1);
from address(7) run sign(2);
from address(7) run performActionEndpoint(1);
from address(7) run performActionEndpoint(2);
