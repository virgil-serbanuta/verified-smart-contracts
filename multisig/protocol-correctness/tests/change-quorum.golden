<T>
  <k>
    .
  </k>
  <state>
    <multisig-state>
      <users>
        <num-users>
          u ( 2 )
        </num-users>
        <user-id-to-address>
          u ( 1 ) |-> address ( 7 )
          u ( 2 ) |-> address ( 8 )
        </user-id-to-address>
        <address-to-user-id>
          address ( 7 ) |-> u ( 1 )
          address ( 8 ) |-> u ( 2 )
        </address-to-user-id>
      </users>
      <board-state>
        <num-board-members>
          u ( 2 )
        </num-board-members>
        <quorum>
          u ( 2 )
        </quorum>
        <num-proposers>
          u ( 0 )
        </num-proposers>
        <user-roles>
          u ( 1 ) |-> BoardMember
          u ( 2 ) |-> BoardMember
        </user-roles>
      </board-state>
      <action-state>
        <action-last-index>
          u ( 2 )
        </action-last-index>
        <actions>
          <action-data>
            u ( 1 ) |-> Nothing
            u ( 2 ) |-> Nothing
          </action-data>
          <action-signers>
            u ( 1 ) |-> [ . ]
            u ( 2 ) |-> [ . ]
          </action-signers>
        </actions>
      </action-state>
    </multisig-state>
    <pseudocode-state>
      <variables>
        .Map
      </variables>
      <stack>
        .List
      </stack>
    </pseudocode-state>
    <external-call-env>
      <caller-address>
        uninitialized
      </caller-address>
    </external-call-env>
  </state>
</T>
